### PR TITLE
Rename sys::Database to sys::Branch

### PR DESCRIPTION
**Pull request by fantix**
Rename `sys::Database` to `sys::Branch`. Also, make the old name an alias of the new name.

~This PR contains the minimal changes to make it work. We still leave behind lots of "database" terms in the code.~

### CODE REVIEW

**Comment by fantix** [comment]
Seems like the `CREATE ALIAS` command failed:

```
CREATE ALIAS sys::Database := sys::Branch;
```

```
RuntimeError: non-abstract object type 'sys::Database' has non-abstract ancestors
```

**Comment by elprans** [comment]
> Seems like the `CREATE ALIAS` command failed:
> 
> ```
> CREATE ALIAS sys::Database := sys::Branch;
> ```
> 
> ```
> RuntimeError: non-abstract object type 'sys::Database' has non-abstract ancestors
> ```

This is not because of `ALIAS`, this is because you also need to rename `class Database` to `class Branch`.

File: edb/schema/database.py [hunk 1]
```diff
@@ -36,22 +36,22 @@
...
 ):
     pass
 
 
-class DatabaseCommandContext(sd.ObjectCommandContext[Database]):
+class DatabaseCommandContext(sd.ObjectCommandContext[Branch]):
     pass
 
 
 class DatabaseCommand(
```
Comment (elprans) [inline edb/schema/database.py:53]: These should be renamed also.

**Review by msullivan — APPROVED** [review]
This is *technically* not backward compatible, as now you can't do thinks like `[is sys::Database]`, but oh well.

(We should maybe support that also)

**Comment by fantix** [comment]
@elprans wdyt?

**Comment by fantix** [comment]
I'm merging this first; we can change it later again.

**Pull request merged**